lib/puppet/pops/time/timespan.rb

```diff
@@ -0,0 +1,686 @@
+module Puppet::Pops
+module Time
+  NSECS_PER_USEC = 1000
+  NSECS_PER_MSEC = NSECS_PER_USEC * 1000
+  NSECS_PER_SEC  = NSECS_PER_MSEC * 1000
+  NSECS_PER_MIN  = NSECS_PER_SEC  * 60
+  NSECS_PER_HOUR = NSECS_PER_MIN  * 60
+  NSECS_PER_DAY  = NSECS_PER_HOUR * 24
+
+  KEY_STRING = 'string'.freeze
+  KEY_FORMAT = 'format'.freeze
+  KEY_NEGATIVE = 'negative'.freeze
+  KEY_DAYS = 'days'.freeze
+  KEY_HOURS = 'hours'.freeze
+  KEY_MINUTES = 'minutes'.freeze
+  KEY_SECONDS = 'seconds'.freeze
+  KEY_MILLI_SECONDS = 'milli_seconds'.freeze
+  KEY_MICRO_SECONDS = 'micro_seconds'.freeze
+  KEY_NANO_SECONDS = 'nano_seconds'.freeze
+
+  # TimeData is a Numeric that stores its value internally as nano-seconds but will be considered to be seconds and fractions of
+  # seconds when used in arithmetic or comparison with other Numeric types, .
+  #
+  class TimeData < Numeric
+    include LabelProvider
+
+    attr_reader :nsecs
+
+    def initialize(nano_seconds)
+      @nsecs = nano_seconds
+    end
+
+    def eql?(o)
+      o.class == self.class ? @nsecs.eql?(o.nsecs) : nil
+    end
+
+    def ==(o)
+      eql?(o)
+    end
+
+    def <=>(o)
+      case o
+      when self.class
+        @nsecs <=> o.nsecs
+      when Integer
+        to_int <=> o
+      when Float
+        to_f <=> o
+      else
+        nil
+      end
+    end
```

Comment:
Yes. A Timespan or Timestamp can be compared to an number that represents seconds (or seconds since epoch) in Puppet.